tests/CMakeLists.txt

```diff
@@ -145,36 +145,49 @@ foreach(testlib_source ${testlibs})
 endforeach()
 add_custom_target(testlibs DEPENDS ${compiled_testlibs})
 
+
+#
+# Runtime Tests
+#
 configure_file(runtime-tests.sh runtime-tests.sh COPYONLY)
-file(GLOB runtime_tests runtime/*)
-list(REMOVE_ITEM runtime_tests ${CMAKE_CURRENT_SOURCE_DIR}/runtime/engine)
-list(REMOVE_ITEM runtime_tests ${CMAKE_CURRENT_SOURCE_DIR}/runtime/scripts)
-list(REMOVE_ITEM runtime_tests ${CMAKE_CURRENT_SOURCE_DIR}/runtime/outputs)
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/runtime/)
-foreach(runtime_test ${runtime_tests})
-  configure_file(${runtime_test} ${CMAKE_CURRENT_BINARY_DIR}/runtime/ COPYONLY)
+add_custom_target(
+  runtime_tests
+  COMMAND ./runtime-tests.sh
+  DEPENDS
+    ${compiled_testprogs}
+    ${compiled_testlibs}
+    ${CMAKE_BINARY_DIR}/src/bpftrace
+)
+add_test(NAME runtime_tests COMMAND ./runtime-tests.sh)
+
+file(GLOB_RECURSE runtime_test_files
+  RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+  CONFIGURE_DEPENDS
```

Comment:
This seems to work but in the docs (https://cmake.org/cmake/help/latest/command/file.html#filesystem) it warns about using glob

Reply:
Yeah I saw that, but since we've been using `GLOB` all over the place for years then this is not a step backwards at least, and `CONFIGURE_DEPENDS` makes it an improvement with most generators.

The other (recommended) option is to explicitly list every source file in a CMakeLists.txt, which sounds like a lot of work 😅